plugins/modules/aws_s3.py

```diff
@@ -674,6 +720,47 @@ def put_download_url(module, s3, bucket, obj, expiry):
     return url
 
 
+def copy_object_to_bucket(module, s3, bucket, obj, encrypt, metadata, validate, d_etag):
+    if module.check_mode:
```

Comment:
Since the `aws_s3` module supports tags, I would suggest that you consider this when copying an object from one bucket to another, whether you keep the same tags or replace them. The same should apply to the other properties. I would also suggest adding some tests asserting that this behavior is assured.